de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/SarosServer.java

```diff
@@ -106,17 +111,24 @@ private void connectToXMPPServer() {
      * 
      * @param args
      *            command-line arguments
-     * @throws Exception
      */
-    public static void main(String[] args) throws Exception {
-        SarosServer server = new SarosServer();
+    public static void main(String[] args) {
+        final SarosServer server = new SarosServer();
 
         LOG.info("Starting server...");
         server.start();
 
-        System.in.read();
-
-        LOG.info("Stopping server...");
-        server.stop();
+        Runtime.getRuntime().addShutdownHook(new Thread(new Runnable() {
+            @Override public void run() {
+                LOG.info("Stopping server...");
+                server.stop();
+            }
+        }));
+
+        if (ServerConfig.isInteractive()) {
+            ServerConsole console = new ServerConsole(System.in, System.out);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this use of System.out or System.err by a logger. ([squid:S106](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS106))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=143&issues=AWZTTLKlBfefbO9HkYcd&open=AWZTTLKlBfefbO9HkYcd)

[ISSUE_KEY]: AWZTTLKlBfefbO9HkYcd